geometry/point3d.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "geometry/latlon.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "base/math.hpp"
+
+#include <cmath>
+
+namespace m3
+{
+template <typename T>
+class Point
+{
+public:
+  constexpr Point() : x(T()), y(T()), z(T()) {}
+  constexpr Point(T const & x, T const & y, T const & z) : x(x), y(y), z(z) {}
+
+  T Length() const { return std::sqrt(x * x + y * y + z * z); }
+
+  T x;
+  T y;
+  T z;
+};
+
+template <typename T>
+T DotProduct(Point<T> const & a, Point<T> const & b)
+{
+  return a.x * b.x + a.y * b.y + a.z * b.z;
+}
+
+template <typename T>
+Point<T> CrossProduct(Point<T> const & a, Point<T> const & b)
+{
+  auto const x = a.y * b.z - a.z * b.y;
+  auto const y = a.z * b.x - a.x * b.z;
+  auto const z = a.x * b.y - b.x * a.y;
```

Comment:
s/b.x * a.y/a.y * b.x/

(it will be more readable and will pass the sanity check that every column is a cyclic shift of "xyz")